### PR TITLE
User signup

### PR DESCRIPTION
**Pull request by mathiasgredal**

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -11,6 +11,7 @@ To run the application on android device/emulator:
 
 ### Browser
 To run the application in a Browser:
+
```
Comment (Nolle10) [inline README.MD:14]: Oooga booga, me see space me likey

CavemanspongebobReactGIF

File: app/src/commonMain/kotlin/sdu/mobile/xpence/ui/components/entryFeild/EmailTextField.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                start = 16.dp,
+                top = 30.dp,
+                end = 16.dp,
+                bottom = 20.dp
+            )
+    )
+}
+private fun isValidText(text: String): Boolean {
+    // Add your custom validation rules here
+    return text.matches(Regex("(?:[a-z0-9!#\$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#\$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])"))
```
Comment (Nolle10) [inline app/src/commonMain/kotlin/sdu/mobile/xpence/ui/components/entryFeild/EmailTextField.kt:53]: Regex go brrrrrrrrrrrrrrrrrrrrr

File: app/build.gradle.kts [hunk 3]
```diff
@@ -102,6 +102,9 @@ compose.experimental {
 libres {
 }
 tasks.getByPath("jsProcessResources").dependsOn("libresGenerateResources")
+dependencies {
```
Comment (Nolle10) [inline app/build.gradle.kts:105]: Delete...

File: build.gradle.kts [hunk 4]
```diff
@@ -1,8 +1,25 @@
...
 plugins {
     id("com.quittle.setup-android-sdk") version "3.1.0"
     alias(libs.plugins.multiplatform).apply(false)
     alias(libs.plugins.compose).apply(false)
     alias(libs.plugins.android.application).apply(false)
     alias(libs.plugins.libres).apply(false)
     alias(libs.plugins.buildConfig).apply(false)
+    kotlin("jvm") version "1.9.20"
+}
+dependencies {
```
Comment (Nolle10) [inline build.gradle.kts:12]: Please delete every apart from the plugins. You may have an older version, It should be changed in the .gitignore

File: build.gradle.kts [hunk 5]
```diff
@@ -1,8 +1,25 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
 plugins {
     id("com.quittle.setup-android-sdk") version "3.1.0"
     alias(libs.plugins.multiplatform).apply(false)
     alias(libs.plugins.compose).apply(false)
     alias(libs.plugins.android.application).apply(false)
     alias(libs.plugins.libres).apply(false)
     alias(libs.plugins.buildConfig).apply(false)
+    kotlin("jvm") version "1.9.20"
```
Comment (Nolle10) [inline build.gradle.kts:10]: And this as well
```suggestion
```

**Review by Nolle10 — CHANGES_REQUESTED** [review]
Please remove everything new from the build files before merging

File: build.gradle.kts [hunk 6]
```diff
@@ -1,3 +1,5 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
```
Comment (Nolle10) [inline build.gradle.kts:1]:
```suggestion
```

File: app/build.gradle.kts [hunk 7]
```diff
@@ -102,6 +102,9 @@ compose.experimental {
...
+    implementation("androidx.core:core-ktx:+")
```
Comment (Nolle10) [inline app/build.gradle.kts:106]:
```suggestion
```

**Review by Nolle10 — CHANGES_REQUESTED** [review]
Still missing a little

File: build.gradle.kts (hunk 6 above)
Comment (Darchie4) [reply build.gradle.kts]: Yes

File: app/build.gradle.kts (hunk 7 above)
Comment (Darchie4) [reply app/build.gradle.kts]: Yes

**Review by Nolle10 — APPROVED** [review]
Looks great, good work all of you.

**Pull request merged**